hoomd/mpcd/test-py/stream_sphere_test.py

```diff
@@ -0,0 +1,197 @@
+            np.testing.assert_array_almost_equal(snap.particles.velocity[2], -1./4.*np.array([-1.,-2.,np.sqrt(11)]))
+            np.testing.assert_array_almost_equal(snap.particles.position[3], [1.92,-1.96,-np.sqrt(8)+0.05])
+            np.testing.assert_array_almost_equal(snap.particles.velocity[3], [-0.8,0.4,0.5])
+
+        # take another step where both particles are streaming only
+        hoomd.run(1)
+        snap = self.s.take_snapshot()
+        if hoomd.comm.get_rank() == 0:
+            np.testing.assert_array_almost_equal(snap.particles.position[0], [2.85,0.895,np.sqrt(6)+0.075])
+            np.testing.assert_array_almost_equal(snap.particles.velocity[0], [-1.,-0.7,0.5])
+            np.testing.assert_array_almost_equal(snap.particles.position[1], [-0.3,-0.3,-0.3])
+            np.testing.assert_array_almost_equal(snap.particles.velocity[1], [-1.,-1.,-1.])
+            np.testing.assert_array_almost_equal(snap.particles.position[2], 0.96*np.array([-1.,-2.,np.sqrt(11)]))
+            np.testing.assert_array_almost_equal(snap.particles.velocity[2], -1./4.*np.array([-1.,-2.,np.sqrt(11)]))
+            np.testing.assert_array_almost_equal(snap.particles.position[3], [1.84,-1.92,-np.sqrt(8)+0.1])
+            np.testing.assert_array_almost_equal(snap.particles.velocity[3], [-0.8,0.4,0.5])
+
+    # test basic stepping behaviour with slip boundary conditions
+    def test_step_slip(self):
+        mpcd.stream.sphere(R=4., boundary="slip")
+
+        # take one step
+        hoomd.run(1)
+        snap = self.s.take_snapshot()
+        if hoomd.comm.get_rank() == 0:
+            np.testing.assert_array_almost_equal(snap.particles.position[0], [2.95,0.965,np.sqrt(6)+0.025])
+            np.testing.assert_array_almost_equal(snap.particles.velocity[0], [1.,0.7,-0.5])
+            np.testing.assert_array_almost_equal(snap.particles.position[1], [-0.1,-0.1,-0.1])
+            np.testing.assert_array_almost_equal(snap.particles.velocity[1], [-1., -1., -1.])
+            np.testing.assert_array_almost_equal(snap.particles.position[2], 0.99*np.array([-1.,-2.,np.sqrt(11)]))
+            np.testing.assert_array_almost_equal(snap.particles.velocity[2], 1./4.*np.array([-1.,-2.,np.sqrt(11)]))
+            np.testing.assert_array_almost_equal(snap.particles.position[3], [2.,-2.,-np.sqrt(8)])
+            np.testing.assert_array_almost_equal(snap.particles.velocity[3], [0.8,-0.4,-0.5])
+
+        # take another step where one particle will now hit the wall
+        hoomd.run(1)
+        snap = self.s.take_snapshot()
+        if hoomd.comm.get_rank() == 0:
+            r1 = np.array([3.,1.,np.sqrt(6)])       # point of contact
+            v1 = np.array([1.,0.7,-0.5])            # velocity before reflection
+            v_ = v1 - 1/8.*np.dot(v1,r1)*r1         # velocity after reflection
+            r_ = r1+v_*0.05                         # position after reflection
```

Comment:
In python, don't put comments on the same line as the variable. These should go on the line above, and you can combine them together if you don't want to use too many lines.

Also, replace `v_` and `r_` with something else, they are confusing why they are single underscore (which is usually used when you need to use a reserved keyword as a variable name). You might do `r1_before`, `v1_before`, `r1_after`, and `v1_after`, for example.